go/app/server.go

```diff
@@ -181,14 +189,35 @@ func (s *Handlers) GetItem(w http.ResponseWriter, r *http.Request) {
 // and stores it in the image directory.
 func (s *Handlers) storeImage(image []byte) (filePath string, err error) {
 	// STEP 4-4: add an implementation to store an image
-	// TODO:
-	// - calc hash sum
-	// - build image file path
-	// - check if the image already exists
-	// - store image
-	// - return the image file path
-
-	return
+	// check if image is empty
+	if len(image) == 0 {
+		return "", errors.New("image is empty")
+	}
+
+	// calc hash sum (this is [32]byte)
+	hash := sha256.Sum256(image)
+	// convert hash to hexadecimal string
+	hashHex := hex.EncodeToString(hash[:]) // 16進数のファイル名に変換
+	// add extension to the file name
+	ext := ".jpg"
```

Comment:
Are we confirming somewhere that the incoming image is `jpg` only? Ideally we should return an error in the `parseAddItemRequest` as well if image is not `jpg`